### PR TITLE
GH-104787: use managed weakrefs in `_asyncio`

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-104787

### CODE REVIEW

**Comment by gvanrossum** [comment]
I'm not sure who is an expert in these APIs. Maybe @markshannon can review?

**Comment by kumaraditya303** [comment]
> I'm not sure who is an expert in these APIs.

Not much to review, see https://github.com/python/cpython/blob/2ef1dc37f02b08536b677dd23ec51541a60effd7/Modules/_testcapi/heaptype.c#L861-L868 as example of usage

File: Modules/_asynciomodule.c [hunk 1]
```diff
@@ -2665,7 +2652,6 @@ static PyType_Slot Task_slots[] = {
     {Py_tp_clear, (inquiry)TaskObj_clear},
     {Py_tp_iter, (getiterfunc)future_new_iter},
     {Py_tp_methods, TaskType_methods},
-    {Py_tp_members, TaskType_members},
```
Comment (itamaro) [inline Modules/_asynciomodule.c:2668]: is this `Py_tp_members` related to managed weakrefs?

Comment (kumaraditya303) [reply Modules/_asynciomodule.c:2668]: That's how `__weaklistoffset__` was set earlier.

**Pull request merged**